cpu/msp430/mtarch.c

```diff
@@ -66,10 +66,10 @@ mtarch_start(struct mtarch_thread *t,
 
   t->sp = &t->stack[MTARCH_STACKSIZE - 1];
 
-  *t->sp = (unsigned short)mt_exit;
+  *t->sp = (unsigned short)(void *)mt_exit;
```

Comment:
I don't know if it's allowed, but we're using it without issues on a 20bit msp430